src/lib/registration/ui/base_system_registration_dialog.rb

```diff
@@ -126,12 +126,19 @@ def product_details_widgets
       def input_widgets
         options = Storage::InstallationOptions.instance
 
+        reg_code = options.reg_code
+        if reg_code.empty?
+          known_reg_codes = Storage::RegCodes.instance.reg_codes
+          base_product_name = SwMgmt.find_base_product["name"]
+          reg_code = known_reg_codes[base_product_name] || ""
```

Comment:
:red_circle: Then here we could use `known_reg_codes[base_product_name] || known_reg_codes[""] || ""`.